components/script/layout_wrapper.rs

```diff
@@ -551,6 +550,18 @@ impl<'le> ServoLayoutElement<'le> {
             (*self.element.unsafe_get()).get_attr_val_for_layout(namespace, name)
         }
     }
+
+    fn mutate_data(&self) -> Option<AtomicRefMut<NodeData>> {
+        self.get_style_data().map(|d| d.borrow_mut())
+    }
+
+    fn get_partial_layout_data(&self) -> Option<&AtomicRefCell<PartialPersistentLayoutData>> {
+        unsafe {
+            self.get_style_and_layout_data().map(|d| {
```

Comment:
Nope the unsafe block is needed because of the deref, I meant (`map(|d| &**d.ptr)`)